lib/msal-angular/src/msal.service.ts

```diff
@@ -316,7 +316,14 @@ export class MsalService extends UserAgentApplication {
 
     public acquireTokenSilent(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<any> {
         return new Promise((resolve, reject) => {
-            super.acquireTokenSilent(scopes, authority, user, extraQueryParameters).then((token: any) => {
+            super.acquireTokenSilent(scopes, authority, user, extraQueryParameters)
+            .catch((error: any) => {
+                if (this.config.popUp) {
+                    return super.acquireTokenPopup(scopes, authority, user, extraQueryParameters)
```

Comment:
Can you please add the else block for the redirect flow.

Reply:
We should also introduce a new flag fallbackToInteractive in the config. This flag will be used only if user wants to fallback to acquireTokenRedirect/popup when acquireTokenSilent Fails.  Whether to fallback to popup/redirect would still be decided by config.poup.